feature-libs/product-configurator/common/components/configurator-cart-entry-info/configurator-cart-entry-info.component.ts

```diff
@@ -1,47 +1,26 @@
-import { Component, Optional } from '@angular/core';
+import { Component } from '@angular/core';
 import { FormControl } from '@angular/forms';
 import { OrderEntry } from '@spartacus/core';
 import { CartItemContext } from '@spartacus/storefront';
-import { EMPTY, Observable, of } from 'rxjs';
+import { Observable, of } from 'rxjs';
 import { CommonConfiguratorUtilsService } from '../../shared/utils/common-configurator-utils.service';
 
 @Component({
   selector: 'cx-configurator-cart-entry-info',
   templateUrl: './configurator-cart-entry-info.component.html',
 })
 export class ConfiguratorCartEntryInfoComponent {
-  // TODO(#11681): make commonConfigUtilsService a required dependency and remove deprecated constructor
-  /**
-   * Default constructor
-   *
-   * @param {CartItemContext} cartItemContext
-   * @param {CommonConfiguratorUtilsService} commonConfigUtilsService
-   */
-  constructor(
-    cartItemContext: CartItemContext,
-    // eslint-disable-next-line @typescript-eslint/unified-signatures
-    commonConfigUtilsService: CommonConfiguratorUtilsService
-  );
-  /**
-   * @deprecated since 3.3
-   */
-  constructor(cartItemContext: CartItemContext);
-
   constructor(
-    // TODO(#10946): make CartItemContext a required dependency and drop fallbacks to `?? EMPTY`.
-    @Optional() protected cartItemContext?: CartItemContext,
-    @Optional()
-    protected commonConfigUtilsService?: CommonConfiguratorUtilsService
+    protected cartItemContext: CartItemContext,
+    protected commonConfigUtilsService: CommonConfiguratorUtilsService
```

Comment:
Can the reviewer double check if it is really fine? 
In line 13 is says to make it required, but we're using it as optional in line 47

line 13 and line 47 from the old changes.

Reply:
I think it is fine. But in line 47, it should be `readonly shouldShowButton$: Observable<boolean> = this.commonConfigUtilsService.isActiveCartContext(this.cartItemContext);`

Reply:
also need to remove `?` from `this.commonConfigUtilsService?` in this class.